tasks/go.py

```diff
@@ -410,17 +410,15 @@ def tidy_all(ctx):
 
 
 @task
-def tidy(ctx, only_modified_packages=False):
-    # TODO: if only_modified_packages then use `go mod tidy -diff` with golang 1.23, probably in a separate check-mod-tidy task
-    # https://github.com/golang/go/issues/27005
-    from tasks import get_modified_packages
-
-    # TODO: Also include packages that import them
-    modules = get_modified_packages(ctx) if only_modified_packages else DEFAULT_MODULES.values()
-
-    for mod in modules:
+def tidy(ctx):
+    # Note: It's currently faster to tidy everything than looking for exactly what we should tidy
+    promises = []
+    for mod in DEFAULT_MODULES.values():
         with ctx.cd(mod.full_path()):
-            ctx.run("go mod tidy")
+            promises.append(ctx.run("go mod tidy", asynchronous=True))
+
+    for promise in promises:
+        promise.join()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why did you create and use the `promises` list since you're not returning it / printing it ?

Reply:
If the program (main thread) stops before the other threads are done they will be killed. Here we wait for all of them to be done first

Reply:
Got it, thanks ! Maybe a short comment to https://docs.pyinvoke.org/en/stable/api/runners.html#invoke.runners.Runner.run would help !

Reply:
Added!